### PR TITLE
Carplay Arrival UI - Removal

### PR DESCRIPTION
**Pull request by JThramer**
Implements #2000. 

The arrival... get it?

* Removing CarPlay arrival UI. 
* Creating delegate method for arrival event. 
* Also renaming navService -> navigationService and taking it public.

To-do: 
- [x] Write a test?

/cc @mapbox/navigation-ios

### CODE REVIEW

File: MapboxNavigation/CarPlayNavigationViewController.swift [hunk 1]
```diff
@@ -21,7 +21,7 @@ public class CarPlayNavigationViewController: UIViewController {
     
     @objc public var drivingSide: DrivingSide = .right
     
-    var navService: NavigationService
+    @objc public var navigationService: NavigationService
```
Comment (frederoni) [inline MapboxNavigation/CarPlayNavigationViewController.swift]: Add some documentation here and add a blurb in the changelog about this and the new delegate method.

File: MapboxNavigation/CarPlayNavigationViewController.swift [hunk 2]
```diff
@@ -21,7 +21,7 @@ public class CarPlayNavigationViewController: UIViewController {
     
     @objc public var drivingSide: DrivingSide = .right
     
-    var navService: NavigationService
```
Comment (frederoni) [inline MapboxNavigation/CarPlayNavigationViewController.swift]: 👍

File: MapboxNavigation/CarPlayNavigationViewController.swift [hunk 3]
```diff
@@ -521,13 +506,19 @@ public protocol CarPlayNavigationDelegate {
...
+     
+     - postcondition: If you return `false` within this method, you must manually advance to the next leg: obtain the value of the `navigationService` and its `NavigationService.routeProgress` property, then increment the `RouteProgress.legIndex` property.
+     - parameter carPlayNavigationViewController: The CarPlay navigation view controller that has arrived at a waypoint.
+     - parameter waypoint: The waypoint that the user has arrived at.
+     - returns: True to automatically advance to the next leg, or false to remain on the now completed leg.
      */
-    @objc func carPlayNavigationViewControllerDidArrive(_ carPlayNavigationViewController: CarPlayNavigationViewController)
+    @objc(carPlaynavigationViewController:didArriveAtWaypoint:)
```
Comment (frederoni) [inline MapboxNavigation/CarPlayNavigationViewController.swift]: nit: s/carPlayn/carPlayN/

File: MapboxNavigation/CarPlayNavigationViewController.swift [hunk 4]
```diff
@@ -498,13 +487,9 @@ extension CarPlayNavigationViewController: StyleManagerDelegate {
...
 extension CarPlayNavigationViewController: NavigationServiceDelegate {
     public func navigationService(_ service: NavigationService, didArriveAt waypoint: Waypoint) -> Bool {
-        if service.routeProgress.isFinalLeg {
-            presentArrivalUI()
-            carPlayNavigationDelegate?.carPlayNavigationViewControllerDidArrive(self)
-        } else {
-            presentWaypointArrivalUI(for: waypoint)
-        }
-        return false
+        let advancesToNextLeg = carPlayNavigationDelegate?.carPlaynavigationViewController?(self, didArriveAt: waypoint) ?? true
```
Comment (flashspys) [inline MapboxNavigation/CarPlayNavigationViewController.swift]: typo

**Comment by akitchen** [comment]
Looks like we're good to go here, yes?

**Pull request merged**